flink-connector/flink/src/main/java/org/apache/gravitino/flink/connector/store/GravitinoCatalogStore.java

```diff
@@ -96,43 +102,37 @@ public boolean contains(String catalogName) throws CatalogException {
     return gravitinoCatalogManager.contains(catalogName);
   }
 
-  private String getGravitinoCatalogProvider(Configuration configuration) {
+  private BaseCatalogFactory catalogFactory(Map<String, String> configuration) {
     String catalogType =
         Preconditions.checkNotNull(
-            configuration.get(CommonCatalogOptions.CATALOG_TYPE),
+            configuration.get(CommonCatalogOptions.CATALOG_TYPE.key()),
             "%s should not be null.",
             CommonCatalogOptions.CATALOG_TYPE);
 
-    switch (catalogType) {
-      case GravitinoHiveCatalogFactoryOptions.IDENTIFIER:
-        return "hive";
-      default:
-        throw new IllegalArgumentException(
-            String.format("The catalog type is not supported:%s", catalogType));
-    }
-  }
-
-  private Catalog.Type getGravitinoCatalogType(Configuration configuration) {
-    String catalogType =
-        Preconditions.checkNotNull(
-            configuration.get(CommonCatalogOptions.CATALOG_TYPE),
-            "%s should not be null.",
-            CommonCatalogOptions.CATALOG_TYPE);
-
-    switch (catalogType) {
-      case GravitinoHiveCatalogFactoryOptions.IDENTIFIER:
-        return Catalog.Type.RELATIONAL;
-      default:
-        throw new IllegalArgumentException(
-            String.format("The catalog type is not supported:%s", catalogType));
-    }
+    ServiceLoader<Factory> serviceLoader = ServiceLoader.load(Factory.class);
+    // Should be only one.
+    return (BaseCatalogFactory)
+        Streams.stream(serviceLoader.iterator())
+            .filter(
+                catalogFactory ->
+                    (catalogFactory instanceof BaseCatalogFactory)
+                        && (catalogFactory.factoryIdentifier().equalsIgnoreCase(catalogType)))
+            .findFirst()
```

Comment:
Could you add a check if there are multi matching catalogFactories?

Reply:
fixed

Reply:
fixed